api/src/api/userRouter.ts

```diff
@@ -215,6 +229,29 @@ export const userRouterFactory = (
       });
   };
 
+  const updateBusinessNameSearchIfNeeded = async (userData: UserData): Promise<UserData> => {
+    if (shouldUpdateBusinessNameSearch(userData)) {
```

Comment:
Might be more readable if this was a guard clause? Idk, tradeoff between fewer layers of nesting, which is more readable, against having to switch the logic to `if (!shouldUpdateBusinessNameSearch)` which is slightly less readable because of the "not". Up to you